bahmni-commons-omod/src/main/java/org/bahmni/module/bahmnicommons/web/v1_0/controller/search/BahmniAppointmentsPatientSearchController.java

```diff
@@ -0,0 +1,87 @@
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.RestUtil;
+import org.openmrs.module.webservices.rest.web.resource.impl.AlreadyPaged;
+import org.openmrs.module.webservices.rest.web.response.IllegalRequestException;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.BaseRestController;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Controller for REST web service access to
+ * the Search resource.
+ * To be used by the Bahmni appointment scheduling module.
+ */
+@Controller
+@RequestMapping(value = "/rest/" + RestConstants.VERSION_1 + "/appointments/search/patient")
+public class BahmniAppointmentsPatientSearchController extends BaseRestController {
+    
+    private PatientService patientService;
+    
+    @Autowired
+    public void setPatientService(PatientService patientService) {
+        this.patientService = patientService;
+    }
+
+    @RequestMapping(method = RequestMethod.GET)
+    @ResponseBody
+    public ResponseEntity<AlreadyPaged<PatientResponse>> search(HttpServletRequest request,
+                                                                HttpServletResponse response) throws ResponseException {
+
+        try {
+            RequestContext requestContext = RestUtil.getRequestContext(request, response);
+            PatientSearchParameters searchParameters = new PatientSearchParameters(requestContext);
+
+            if (searchParameters.getCustomAttribute() != null ||
+                    searchParameters.getAddressFieldValue() != null ||
+                    searchParameters.getPatientAttributes() != null ||
+                    searchParameters.getProgramAttributeFieldValue() != null ||
+                    searchParameters.getProgramAttributeFieldName() != null ||
+                    searchParameters.getAddressSearchResultFields() != null ||
+                    searchParameters.getPatientSearchResultFields() != null ||
+                    searchParameters.getLoginLocationUuid() != null) {
+                throw new IllegalRequestException("An unsupported search parameter was provided.");
+            }
+
+            String query = searchParameters.getIdentifier() != null ? searchParameters.getIdentifier() : searchParameters.getName();
+            List<Patient> patients = patientService.getPatients(query);
```

Comment:
@icrc-psousa a comment [here](https://bahmni.slack.com/archives/C03LWT96L/p1604463400114500?thread_ts=1604333125.112800&cid=C03LWT96L) on Slack to double check maybe?